extensions/src/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java

```diff
@@ -71,7 +71,17 @@ public UserListWrapper getUsers() {
   }
 
   public RoleListWrapper getRoles() {
-    return new RoleListWrapper( getUserRoleListService().getAllRoles() );
+    return getRoles( true );
+  }
+
+  public RoleListWrapper getRoles( boolean includeExtraRoles ) {
+    List<String> roles = getUserRoleListService().getAllRoles();
+    if ( !includeExtraRoles ) {
+      for ( String role : getExtraRoles() ) {
+        roles.remove( role );
```

Comment:
Moderately concerned here - if the "extra" role is also not an extra role (comes in from LDAP), then this will remove it anyway. This may be why we designed this stuff to have them as separate items.

From LDAP:
abc
def
Authenticated
ghi

Extra Roles
Authenticated
Administration

getRoles( false ):
abc
def
ghi

Is that correct ?